src/test/java/seedu/address/logic/parser/MarkingCommandParserTest.java

```diff
@@ -42,34 +44,51 @@ public void parse_validArgs_returnsMarkingCommand() {
+
         // Not a number
         assertParseFailure(parserForMark, "a",
-                String.format(MESSAGE_INVALID_COMMAND_FORMAT, MarkCommand.MESSAGE_USAGE));
+                String.format(errorDueToInvalidIndex, MarkCommand.MESSAGE_USAGE));
         assertParseFailure(parserForUnmark, "a",
-                String.format(MESSAGE_INVALID_COMMAND_FORMAT, UnmarkCommand.MESSAGE_USAGE));
+                String.format(errorDueToInvalidIndex, UnmarkCommand.MESSAGE_USAGE));
 
         // Zero index
         assertParseFailure(parserForMark, "0",
-                String.format(MESSAGE_INVALID_COMMAND_FORMAT, MarkCommand.MESSAGE_USAGE));
+                String.format(errorDueToInvalidIndex, MarkCommand.MESSAGE_USAGE));
         assertParseFailure(parserForUnmark, "0",
-                String.format(MESSAGE_INVALID_COMMAND_FORMAT, UnmarkCommand.MESSAGE_USAGE));
+                String.format(errorDueToInvalidIndex, UnmarkCommand.MESSAGE_USAGE));
 
         // Negative index
         assertParseFailure(parserForMark, "-1",
-                String.format(MESSAGE_INVALID_COMMAND_FORMAT, MarkCommand.MESSAGE_USAGE));
+                String.format(errorDueToInvalidIndex, MarkCommand.MESSAGE_USAGE));
         assertParseFailure(parserForUnmark, "-1",
-                String.format(MESSAGE_INVALID_COMMAND_FORMAT, UnmarkCommand.MESSAGE_USAGE));
+                String.format(errorDueToInvalidIndex, UnmarkCommand.MESSAGE_USAGE));
 
         // Multiple inputs, one invalid
         assertParseFailure(parserForMark, "0 1",
-                String.format(MESSAGE_INVALID_COMMAND_FORMAT, MarkCommand.MESSAGE_USAGE));
+                String.format(errorDueToInvalidIndex, MarkCommand.MESSAGE_USAGE));
         assertParseFailure(parserForUnmark, "0 1",
-                String.format(MESSAGE_INVALID_COMMAND_FORMAT, UnmarkCommand.MESSAGE_USAGE));
+                String.format(errorDueToInvalidIndex, UnmarkCommand.MESSAGE_USAGE));
 
         // Multiple inputs, all invalid
         assertParseFailure(parserForMark, "0 -1",
-                String.format(MESSAGE_INVALID_COMMAND_FORMAT, MarkCommand.MESSAGE_USAGE));
+                String.format(errorDueToInvalidIndex, MarkCommand.MESSAGE_USAGE));
         assertParseFailure(parserForUnmark, "0 -1",
-                String.format(MESSAGE_INVALID_COMMAND_FORMAT, UnmarkCommand.MESSAGE_USAGE));
+                String.format(errorDueToInvalidIndex, UnmarkCommand.MESSAGE_USAGE));
+
+        // Multiple inputs, all valid, duplicates
+        assertParseFailure(parserForMark, "1 1",
+                String.format(errorDueToDuplicateIndex, MarkCommand.MESSAGE_USAGE)); // Only 2 inputs, duplicates
+        assertParseFailure(parserForUnmark, "1 1",
+                String.format(errorDueToDuplicateIndex, UnmarkCommand.MESSAGE_USAGE)); // Only 2 inputs, duplicates
+        assertParseFailure(parserForMark, "70 2 3 2 2 1 70",
```

Comment:
I think you can add a comment to tell that the duplicate index error will be thrown before the invalid range error